lib/ivis_opengl/pieblitfunc.cpp

```diff
@@ -128,18 +127,24 @@ void GFX::updateTexture(const void *image, int width, int height)
 
 void GFX::buffers(int vertices, const GLvoid *vertBuf, const GLvoid *auxBuf)
 {
-	glBindBuffer(GL_ARRAY_BUFFER, mBuffers[VBO_VERTEX]);
-	glBufferData(GL_ARRAY_BUFFER, vertices * mCoordsPerVertex * sizeof(GLfloat), vertBuf, GL_STATIC_DRAW);
+	if (mBuffers[VBO_VERTEX])
+		delete mBuffers[VBO_VERTEX];
+	mBuffers[VBO_VERTEX] = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, vertices * mCoordsPerVertex * sizeof(gfx_api::gfxFloat));
```

Comment:
This is not equivalent code to the original code, and will introduce a performance regression. You are creating and deleting the buffer every time through. There is overhead to that.